gtdynamics/optimizer/Optimizer.h

```diff
@@ -54,5 +62,16 @@ class Optimizer {
    */
   gtsam::Values optimize(const gtsam::NonlinearFactorGraph& graph,
                          const gtsam::Values& initial_values) const;
+
+  /**
+   * @brief optimize with constraints using optimizer settings.
+   *
+   * @param graph a Nonlinear factor graph built by derived class
+   * @param initial_values Initial values for all variables.
+   * @return Values The result of the optimization.
+   */
+  gtsam::Values optimize(const gtsam::NonlinearFactorGraph& graph,
```

Comment:
Should be the only optimizer.

Reply:
Keep the old one for compatibility of old code.

Reply:
Add TODO to remove then after discussing with team